CactbotOverlay/FFXIVProcess.cs

```diff
@@ -2,1261 +2,1559 @@
+      //     0x014 bytes in: int16 maxgp;
+      //     0x016 bytes in: int16 cp;
+      //     0x018 bytes in: int16 maxcp;
+      //     ...
+      //     0x03C bytes in: EntityJob job;  // 1 byte.
+      //     ...
+      //     0x03E bytes in: int8 level;
+      // }
+
+      [StructLayout(LayoutKind.Explicit)]
+      public unsafe struct EntityMemory {
+         public static int Size => Marshal.SizeOf(typeof(EntityMemory));
+
+         // Unknown size, but this is the bytes up to the next field.
+         public const int nameBytes = 68;
+
+         [FieldOffset(0x30)]
+         public fixed byte Name[nameBytes];
+
+         [FieldOffset(0x74)]
+         public uint id;
+
+         [FieldOffset(0x8C)]
+         public EntityType type;
+
+         [FieldOffset(0x92)]
+         public ushort distance;
+
+         [FieldOffset(0xA0)]
+         public Single pos_x;
+
+         [FieldOffset(0xA4)]
+         public Single pos_y;
+
+         [FieldOffset(0xA8)]
+         public Single pos_z;
+
+         [FieldOffset(0x18B8)]
+         public CharacterDetails charDetails;
+
+      }
+
+      [StructLayout(LayoutKind.Explicit)]
+      public struct CharacterDetails {
+
+         [FieldOffset(0x00)]
+         public int hp;
+
+         [FieldOffset(0x04)]
+         public int max_hp;
+
+         [FieldOffset(0x08)]
+         public int mp;
+
+         [FieldOffset(0x0C)]
+         public int max_mp;
+
+         [FieldOffset(0x12)]
+         public short gp;
+
+         [FieldOffset(0x14)]
+         public short max_gp;
+
+         [FieldOffset(0x16)]
+         public short cp;
+
+         [FieldOffset(0x18)]
+         public short max_cp;
+
+         [FieldOffset(0x3C)]
+         public EntityJob job;
+
+         [FieldOffset(0x3E)]
+         public short level;
+
+         [FieldOffset(0x5F)]
+         public short shieldPercentage;
+      }
+
+      // A piece of code that reads the job data.
+      // The pointer of interest is the first ???????? in the signature.
+      private static String kJobDataSignature = "488B0D????????4885C90F84????????488B05????????3C03";
+      private static int kJobDataSignatureOffset = -22;
+      // The signature finds a pointer in the executable code which uses RIP addressing.
+      private static bool kJobDataSignatureRIP = true;
+
+      // The op before the pointer wildcard in the signature reads a pointer-to-a-pointer
+      // to the job-specific data structure. We call it |outer| below:
+      //
+      // JobDataOuterStruct* outer;  // This pointer found from the signature.
+      // JobDataOuterStruct {
+      //   JobDataInnerStruct* inner;  // This points to the address after it currently.
+      //   JobDataInnerStruct {
+      //      SomeJobSpecificStruct* struct;  // This is a pointer that changes when you change job, and points to a lot of other pointers.
+      //      union JobSpecificData {
+      //        struct RedMage {
+      //          0x8 bytes in: byte white_mana;
```

Comment:
I think all of these comments can be removed as well, as the layout explicit self-documents.